react/HomeOpenGraph.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Fragment } from 'react'
+import {
+  Helmet,
+  useRuntime,
+  RenderContext,
+  canUseDOM,
+} from 'vtex.render-runtime'
+
+// eslint-disable-next-line no-var
+declare var global: {
+  __hostname__: string
+  __pathname__: string
+}
+
+interface MetaTag {
+  property: string
+  content: string
+}
+
+function HomeOpenGraph() {
+  const runtime = useRuntime() as RenderContext
+
+  const { getSettings } = runtime
+
+  const hostname = canUseDOM ? window.location.hostname : global.__hostname__
+  const pathname = canUseDOM ? window.location.pathname : global.__pathname__
+  const url = `https://${hostname}${pathname}`
+  let title = ''
+  let description = ''
+
+  try {
+    const settings = getSettings('vtex.store')
+
+    if (settings) {
+      const { storeName, titleTag, metaTagDescription } = settings
+
+      title = titleTag ?? storeName
+      description = metaTagDescription
+    }
+  } catch (e) {
+    console.error('Failed to load store settings.', e)
+  }
+
+  // TODO: Add og:image tag, which should be customizable via admin
+  const metaTags: MetaTag[] = [
+    { property: 'og:type', content: 'website' },
+    { property: 'og:title', content: title },
+    { property: 'og:url', content: url },
+    { property: 'og:description', content: description },
+  ]
```

Comment:
Use `useMemo` maybe?

Reply:
In this case, `useMemo` would not give us much, since this is a simple assignment. I mean, I know there is the creation of new objects, but that's not really a big deal in this case. `useMemo` would be useful if there were some computations happening.